adcgen/expression/object_container.py

```diff
@@ -724,6 +731,81 @@ def rename_tensor(self, current: str, new: str,
             obj = ExprContainer(obj, **self.assumptions)
         return obj
 
+    def expand_coulomb_ri(self, factorisation: str = 'sym',
+                          wrap_result: bool = True) -> "ExprContainer | Expr":
+        """
+        Expands the Coulomb operators (pq | rs) into RI format
+
+        Parameters
+        ----------
+        factorisation : str, optional
+            The type of factorisation ('sym' or 'asym'), by default 'sym'
+        wrap_result : bool, optional
+            Whether to wrap the result in an ExprContainer, by default True
+
+        Returns
+        -------
+        ExprContainer | Expr
+            The factorised expression.
+
+        Raises
+        ------
+        NotImplementedError
+            If a factorisation that is not 'sym' or 'asym' is provided or if
+            the expression is not real.
+        ValueError
+            If an invalide exponent is present
+        """
+        from .expr_container import ExprContainer
+
+        if factorisation not in ('sym', 'asym'):
+            raise NotImplementedError("Only symmetric (sym) and asymmetric "
+                                      "(asym) factorisation of the Coulomb "
+                                      "integral is implemented")
+
+        res = self.inner
+        base, exponent = self.base_and_exponent
+        if not exponent.is_Integer:
```

Comment:
The check for integer exponents should be shifted to the if clause below as it is only relevant for the coulomb tensors that we want expand here - we don't care about the exponents of other tensors.